docs/guides/tfhers/shared-key.md

```diff
@@ -0,0 +1,247 @@
+
+```python
+from functools import partial
+from concrete.fhe import tfhers
+
+tfhers_params = tfhers.CryptoParams(
+    lwe_dimension=909,
+    glwe_dimension=1,
+    polynomial_size=4096,
+    pbs_base_log=15,
+    pbs_level=2,
+    lwe_noise_distribution=9.743962418842052e-07,
+    glwe_noise_distribution=2.168404344971009e-19,
+    encryption_key_choice=tfhers.EncryptionKeyChoice.BIG,
+)
+# creating a TFHE-rs ciphertext type with crypto and encoding params
+tfhers_type = tfhers.TFHERSIntegerType(
+    is_signed=False,
+    bit_width=8,
+    carry_width=3,
+    msg_width=2,
+    params=tfhers_params,
+)
+# this partial will help us create TFHERSInteger with the given type instead of calling
+# tfhers.TFHERSInteger(tfhers_type, value) every time
+tfhers_int = partial(tfhers.TFHERSInteger, tfhers_type)
+```
+
+## Defining the Circuit and Compiling
+
+We will now define a simple modular addition function. This function takes TFHE-rs inputs, converts them to Concrete format (`to_native`), runs a computation, and then converts them back to TFHE-rs. The circuit below is a common example that takes and produces TFHE-rs ciphertexts. However, there are other scenarios where you might not convert back to TFHE-rs, or you might convert to a different type than the input. Another possibility is to take one native ciphertext and one TFHE-rs ciphertext.
+
+```python
+def compute(tfhers_x, tfhers_y):
+    ####### TFHE-rs to Concrete #########
+
+    # x and y are supposed to be TFHE-rs values.
+    # to_native will use type information from x and y to do
+    # a correct conversion from TFHE-rs to Concrete
+    concrete_x = tfhers.to_native(tfhers_x)
+    concrete_y = tfhers.to_native(tfhers_y)
+    ####### TFHE-rs to Concrete #########
+
+    ####### Concrete Computation ########
+    concrete_res = (concrete_x + concrete_y) % 213
+    ####### Concrete Computation ########
+
+    ####### Concrete to TFHE-rs #########
+    tfhers_res = tfhers.from_native(
+        concrete_res, tfhers_type
+    )  # we have to specify the type we want to convert to
+    ####### Concrete to TFHE-rs #########
+    return tfhers_res
+```
+
+We can compile the circuit as usual.
+
+```python
+compiler = fhe.Compiler(compute, {"tfhers_x": "encrypted", "tfhers_y": "encrypted"})
+inputset = [(tfhers_int(120), tfhers_int(120))]
+circuit = compiler.compile(inputset)
+```
+
+You could optionally try the full execution in Concrete
+
+```python
+# encode/encrypt
+encrypted_x, encrypted_y = circuit.encrypt(tfhers_type.encode(7), tfhers_type.encode(9))
+# run
+encrypted_result = circuit.run(encrypted_x, encrypted_y)
+# decrypt
+result = circuit.decrypt(encrypted_result)
+# decode
+decoded = tfhers_type.decode(result)
+```
+
+## The Bridge!
+
+We are going to create a TFHE-rs bridge that will assist in moving ciphertexts and keys between the two sides.
```

Comment:
We are going to create a TFHE-rs bridge that facilitates the seamless transfer of ciphertexts and keys between Concrete and TFHE-rs.